test/input-states-mocha.js

```diff
@@ -0,0 +1,333 @@
+               inStates.push( { portInfo: inPortInfo, state: inState } ); 
+          } 
+
+          // Tack one onto the front 
+          inPortInfo = { inportName: 'input' };
+          inState = { vnid: 99,
+                      data: 'alpha data',
+                      lm: createLm() };
+          node.vni().inputStates({'input': inState});
+          inStates.push( { portInfo: inPortInfo, state: inState } ); 
+
+          // Insert one in the middle 
+          inPortInfo = { inportName: 'input1' };
+          inState = { vnid: 22,
+                      data: 'middle data',
+                      lm: createLm() };
+          node.vni().inputStates({'input1': inState});
+          inStates.push( { portInfo: inPortInfo, state: inState } ); 
+
+          // Verify that we have all of them 
+          inStates.forEach( function(inState) { 
+              var actualState = node.vni().inputStates(inState.portInfo.inportName);
+              actualState.should.deep.equal( inState.state );
+          });
+    });
+    it('should fall back to no vnid state', function() {
+         var state = { vnid: '010',
+                       data: 'hello world from originating node',
+                       lm: createLm() }
+
+         var context = node.vni().inputStates('inputa', 0, state);
+         context.should.exist;
+
+         var result = node.vni('010').inputStates('inputa', 0); 
+         verifyInputState( result, state );
+    }); 
+    it('should set and get a new packet input state', function() {
+         var state = { vnid: '010',
+                       data: 'hello world from originating node',
+                       lm: createLm() }
+
+         var context = node.vni().inputStates('inputa', 0, state);
+         context.should.exist;
+
+         var result = node.vni().inputStates('inputa', 0); 
+         verifyInputState( result, state );
+    }); 
+    it('should set and get a new packet input state by socketIndex', function() {
+         var state = { vnid: '010',
+                       data: 'hello world from originating node',
+                       lm: createLm() }
+
+         var context = node.vni().inputStates('inputa', 0, state);
+         context.should.exist;
+
+         verifyInputState(node.vni().inputStates('inputa', 0), state);
+         node.vni().inputStates('inputa', 0, undefined);
+         expect(node.vni().inputStates('inputa', 0)).to.be.undefined;
+    }); 
+    it('should set and get array of input states', function() {
+         var one = { vnid: '010',
+                       data: 'hello world from node one',
+                       lm: createLm() }
+         var two = { vnid: '010',
+                       data: 'hello world from node two',
+                       lm: createLm() }
+
+         var context = node.vni().inputStates({'inputa': [one, two]});
+         context.should.exist;
+
+         node.vni().inputStates('inputa').should.eql([one, two]);
```

Comment:
This seems a bit odd.  You are setting two states with the same VNID in an array .  Since the vnid is the same for both and on the same port, wouldn't state two overwrite one?   Please add a comment as to what you are doing with this mix.